theano/tensor/nnet/corr_gemm.c

```diff
@@ -172,50 +171,61 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
     }
 
     // Create temporary columns
-    npy_intp col_dim[2];
-    col_dim[0] = (npy_intp)(nChannels * kW * kH);
-    col_dim[1] = (npy_intp)(topHeight * topWidth);
-    PyArrayObject* col = (PyArrayObject*)PyArray_EMPTY(2,
-		                           col_dim,
-                                           PyArray_TYPE(top),
-					   0);
+    const int max_threads = %(omp_max_threads)s < batchSize ? %(omp_max_threads)s : batchSize;
+
+    npy_intp col_dim[3];
+    col_dim[0] = (npy_intp)max_threads;
+    col_dim[1] = (npy_intp)(nChannels * kW * kH);
+    col_dim[2] = (npy_intp)(topHeight * topWidth);
+    PyArrayObject* col = (PyArrayObject*)PyArray_ZEROS(3,
```

Comment:
@nouiz looks like PyArray_ZEROS could run with multi-thread mode in Numpy, while PyArray_EMPTY can't. So I got the better result with PyArray_ZEROS when `max_threads > 1`.
 
Here's the snippet of profiling of AlexNet Workload when `max_threads>1`:

**Calling PyArray_ZEROS**
```
 <% time> <sum %> <apply time> <time per call> <type> <#call> <#apply> <Class name>
   19.7%    40.1%      23.662s       1.18e-01s     C      200      15   theano.tensor.nnet.corr.CorrMM
   10.3%    67.1%      12.288s       1.23e-01s     C      100       5   theano.tensor.nnet.corr.CorrMM_gradWeights
    9.5%    76.6%      11.422s       1.43e-01s     C       80       4   theano.tensor.nnet.corr.CorrMM_gradInputs
```

**Calling PyArray_EMPTY**
```
 <% time> <sum %> <apply time> <time per call> <type> <#call> <#apply> <Class name>
  23.1%    23.1%      30.808s       1.54e-01s     C      200      15   theano.tensor.nnet.corr.CorrMM
  12.2%    68.4%      16.244s       1.62e-01s     C      100       5   theano.tensor.nnet.corr.CorrMM_gradWeights
  10.7%    79.0%      14.218s       1.78e-01s     C       80       4   theano.tensor.nnet.corr.CorrMM_gradInputs
```